### PR TITLE
Maglev Rails

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Maglev cars can only move on turfs that have a maglev rail.

Maglev cars move at max speed.

It costs 1 plasteel sheet to build a rail section.

TODO:

Sprites

Probably make maglev tracks its own stack item so it can be a bit cheaper

Add a way to get the maglev car itself.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
halflife train cars are go?

**Comment by Incoming5643** [comment]
How is this any different than transit tubes

**Comment by PKPenguin321** [comment]
its cooler and hopefully it will be less clunky

**Comment by KorPhaeron** [comment]
I guess it is sorta redundant. I think the biggest differences for these are that

1. The passenger is exposed to being attacked, and can attack in turn.

2. The passenger can steer/start/stop when they want to.

3. It's far, far simpler to place these down

4. The code is hopefully way less horrifying

**Comment by Bawhoppen** [comment]
we have rail sprites in mining icons file, will those work?

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -428,3 +428,43 @@
...
+/obj/vehicle/maglev/relaymove(mob/user, direction)
+	var/turf/t = get_step(src, direction)
+	var/obj/structure/maglev/N = locate() in t
+	if(N)
+		..()
+	else
+		user << "You need a maglev rail to move!"
+		return 0
+
+/obj/vehicle/maglev/keycheck()
```
Comment (RemieRichards) [inline code/modules/mining/mine_items.dm]: this is unneeded, no ```keytype``` means they automatically return 1 here already.

Comment (KorPhaeron) [inline code/modules/mining/mine_items.dm]: Whoops

**Comment by KorPhaeron** [comment]
@Bawhoppen 

Those are

1) Not smoothed

2) Wooden minecart tracks, not maglev rails

**Comment by Bawhoppen** [comment]
how was I supposed to know what maglev rails were before googling it
also make the rail sections cost 2 instead of one

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -428,3 +428,43 @@
 	anchored = 1
 	layer = MOB_LAYER - 0.2
 	density = 0
+
+
+//Maglev Tracks
+
+
+/obj/vehicle/maglev
```
Comment (Fox-McCloud) [inline code/modules/mining/mine_items.dm]: these should probably be buildable, too.

**Comment by Fox-McCloud** [comment]
Pics, please?

I really like the idea; it's pretty neat, though I think the vehicle, itself, should be craftable.

**Comment by KorPhaeron** [comment]
@Fox-McCloud 

See the TODO list

**Comment by Fox-McCloud** [comment]
@KorPhaeron 

Awesome, thanks. Definitely looking forward to this.

**Comment by CosmicScientist** [comment]
:+1:

**Comment by Bawhoppen** [comment]
if you make this a stack then make a whole new file for stacks and name it like "utility" and move my sandbags there

**Comment by FantasticFwoosh** [comment]
It would be very sweet if these had a LAVA IMMUNE tag attached so miners could path shortcuts across rivers to deposits and load ore boxes inside the riding component as a high tech minecart. (assuming that using the same logic as lava boats they wouldn't catch fire)

**Comment by KorPhaeron** [comment]
Still no sprites dead PR

**Comment by FantasticFwoosh** [comment]
Hopefully this will get a new lease of life with the dwarves.

**Comment by PKPenguin321** [comment]
Would also be cool for planetstation

**Comment by KorPhaeron** [comment]
Need sprites still

**Comment by Jordie0608** [comment]
Just lift them from OpenTTD ( ͡° ͜ʖ ͡°)

**Pull request closed without merging**